torch/_inductor/fx_passes/post_grad.py

```diff
@@ -47,6 +48,91 @@
 inference_patterns = PatternMatcherPass()
 
 
+def get_storage(t):
+    return t.untyped_storage()._cdata
+
+
+def get_node_storage(node):
+    if not isinstance(node.meta["val"], torch.Tensor):
+        return None
+    if not torch._C._has_storage(node.meta["val"]):
+        return None
+    return get_storage(node.meta["val"])
+
+
+class FakeTensorUpdater:
+    def __init__(self, graph):
+        self.processed_hashes = set()
+        self.graph = graph
+
+        for node in self.graph.nodes:
+            self.processed_hashes.add(self.hash_node(node))
+
+    def hash_node(self, node):
+        return (node, node.target, node.args, node.kwargs)
+
+    def incremental_update(self):
+        """
+        Not fully featured yet. Mainly added to ensure that reinplacing pass is
+        sound.
+
+        Todo things:
+        1. We probably want to persist this across more passes.
+        """
+        return
+        processed = set()
+        existing_tensors = defaultdict(int)
+        for node in self.graph.nodes:
+            existing_tensors[get_node_storage(node)] += 1
+
+        def is_fake_tensor_same(new, old):
+            if new.shape != old.shape or new.stride() != old.stride():
+                return False
+            if get_storage(new) == get_storage(old):
+                return True
+            if (
+                existing_tensors[get_storage(old)] == 1
+                and get_storage(new) not in existing_tensors
+            ):
+                return True
+            return False
+
+        for node in self.graph.nodes:
+            if self.hash_node(node) in old_hash:
+                continue
+
+            def get_fake_tensor(x):
+                if isinstance(x, torch.fx.Node):
+                    return x.meta["val"]
+                return x
+
+            if node.op == "call_function" and isinstance(
+                node.target, torch._ops.OpOverload
+            ):
+                processing = [node]
+                while len(processing) > 0:
+                    updating_node = processing.pop()
+                    if updating_node in processed:
+                        continue
+                    if node.op != "call_function" or not isinstance(
+                        node.target, torch._ops.OpOverload
+                    ):
+                        continue
+
+                    args, kwargs = tree_map(
+                        get_fake_tensor, (updating_node.args, updating_node.kwargs)
+                    )
+                    new_fake_tensor = updating_node.target(*args, **kwargs)
```

Comment:
Need to enable fake mode, wont work with constructors. Maybe it's already enabled ?